### PR TITLE
Add LiveRamp identifier

### PR DESCRIPTION
**Pull request by ikp4success**
*Issue #, if available:*
#40 

*Description of changes:*
Add LiveRamp identifier support

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

image

### CODE REVIEW

File: source/api/tests/test_api_integration.py
Comment (ianwow) [inline source/api/tests/test_api_integration.py:413]: A ramp id is not going to be a "product_name". This would be a better column description:

"name": "ramp_id",
"description": "The user's LiveRamp ID",
"dataType": "STRING",
"nullable": true,
"externalUserIdType": {
    "type": "LiveRamp"
}

File: source/api/tests/test_api_integration.py
Comment (ianwow) [inline source/api/tests/test_api_integration.py:423]: Did you misname this test, or forget to add a live ramp column to test_columns?

File: source/api/tests/test_api_integration.py
Comment (ianwow) [inline source/api/tests/test_api_integration.py:456]: Same comment as above. A live ramp column will be a live ramp id, not a product quantity.

File: source/website/src/views/Step3.vue
Comment (ianwow) [inline source/website/src/views/Step3.vue:236]: This would be more consistent with how live ramp is described in the AMC upload docs:

{ value: 'LiveRamp ID', text: 'ramp_id disabled: false},

File: source/website/src/views/Step3.vue
Comment (ianwow) [inline source/website/src/views/Step3.vue:370]: Why this change? It does not appear to relate to liveramp.

File: source/website/src/views/Step3.vue
Comment (ianwow) [inline source/website/src/views/Step3.vue:388]: We shouldn't be trying to support two naming conventions. If you want to use a different convention, then rename the old vars.

Comment (ikp4success) [reply source/website/src/views/Step3.vue:388]: ok i will remove this.

File: source/website/src/views/Step3.vue
Comment (ianwow) [inline source/website/src/views/Step3.vue:401]: add a comment to describe what this function does

File: source/website/src/views/Step3.vue
Comment (ianwow) [inline source/website/src/views/Step3.vue:488]: There can only be one live ramp id. Once someone selects a ramp id column type then the should not be able to assign another column that column type (i.e. grey out that menu option).

File: source/api/tests/test_api_integration.py
Comment (ikp4success) [reply source/api/tests/test_api_integration.py]: Resolved

File: source/website/src/views/Step3.vue
Comment (ikp4success) [reply source/website/src/views/Step3.vue]: Resolved

File: source/website/src/views/Step3.vue
Comment (ikp4success) [reply source/website/src/views/Step3.vue]: Resolved

File: source/website/src/views/Step3.vue
Comment (ikp4success) [reply source/website/src/views/Step3.vue]: Resolved

File: source/website/src/views/Step3.vue
Comment (ikp4success) [reply source/website/src/views/Step3.vue]: Resolved

File: source/website/src/views/Step3.vue
Comment (ikp4success) [reply source/website/src/views/Step3.vue]: Resolved

File: source/api/tests/test_api_integration.py
Comment (ikp4success) [reply source/api/tests/test_api_integration.py]: Resolved

File: source/api/app.py
Comment (ianwow) [inline source/api/app.py:148]: Did you accidentally merge code from the "country code" feature into this branch?

**Review by ianwow — CHANGES_REQUESTED** [review]
Remove "country code" related changes, or if you intentionally added new support for country code then explain what you did in the PR description and reference the corresponding issue.

File: source/api/app.py
Comment (ikp4success) [reply source/api/app.py:148]: Its pre-commit change for sonarcube

Comment (ikp4success) [reply source/api/app.py]: i re-merge development to this branch, so this code is outdated.

**Review by ianwow — CHANGES_REQUESTED** [review]
I still think you're generating an invalid dataset definition. Please include a screenshot that shows what happens when you click submit on step 4 for a dataset that includes a ramp_id column.

**Comment by ikp4success** [comment]
> I still think you're generating an invalid dataset definition. Please include a screenshot that shows what happens when you click submit on step 4 for a dataset that includes a ramp_id column.

image

**Comment by ikp4success** [comment]
@ianwow Step 5
image

File: source/website/src/views/Step3.vue
Comment (ianwow) [inline source/website/src/views/Step3.vue:344]: I recommend using `text: 'LiveRamp ID'` instead of `text: 'ramp_id'`, since that is the text users will actually see.

**Review by ianwow — APPROVED** [review]
I recommend using text: 'LiveRamp ID' instead of text: 'ramp_id', since that is the text users will actually see.

Please don't merge this until I have a chance to merge the `multi_instance_upload2` branch.

**Pull request merged**
